### DISCUSSION TITLE
Cannot figure out public access

### DISCUSSION THREAD

**Discussion by benjamindaines** [immich-app/immich#7600] [General]
Can someone explain to me how to get public access (share links specifically, as I use a VPN to connect my devices)?  I am running opnsense on my router.  I have a sub domain to points to my IP address, however it is not just for immich, it's also the address that I use to connect to my VPN (meaning I cannot just forward all subdomain traffic to Immich).  I've tried to use the HAProxy plugin GUI, but could not get it working.  I even tried just forwarding port 2283 and that didn't even work.  Immich is installed on a system behind the opnsense box, not on the same machine (local IP is 192.188.0.2 for the server). 

Extremely frustrated.  Tempted to just trash the entire docker container.  Help would be greatly appreciated.  The documentation is not clear to me either. 

Thanks.  I need step by step instructions for an idiot.

**Comment by TheGorf** [comment]
If you are using HAProxy, then you need to make sure that you are routing connections to the server via the host header. HAProxy can then read the header and decide what website the client is trying to reach. 

I love HAProxy, but it can be a little finicky. You might be a little better off spinning up something behind your firewall and running 
https://nginxproxymanager.com/
That has a tidy web gui for setting up sites and the Let's Encrypt. 

The second thing you have to fix is NAT. If you are going to hit a site whose IP is the public IP of your firewall from inside your firewall you are going to have asymmetric routing and that's bad. Your firewall has to nat the inbound connection. Probably a better topic for over in the OPNSense forums.

**Comment by thariq-shanavas** [comment]
We don't have step-by-step instructions as the steps would vary wildly by your system. You can find an overview of general concepts here: https://immich.app/docs/guides/remote-access/

Do not expose port 2283 to the internet.
